service/s3/internal/configtesting/go.mod

```diff
@@ -0,0 +1,9 @@
+module github.com/aws/aws-sdk-go-v2/service/s3/internal/configtesting
```

Comment:
What was your reasoning for wanting to keep this separate?

Reply:
This depends on the `config` module to assert that the s3 external config resolvers are valid interface.  If It was not separate then the S3 package would depend on the `config` module, which we probably don't want. This is similar reason `integration` module under `service/dynamodb` module are a broken out as well.